host-controller/src/main/java/org/jboss/as/host/controller/mgmt/HostControllerRegistrationHandler.java

```diff
@@ -204,6 +205,22 @@ public HostControllerRegistrationHandler(ManagementChannelHandler handler, Domai
          */
         ModelNode executeReadOnly(ModelNode operation, Resource model, OperationStepHandler handler, ModelController.OperationTransactionControl control);
 
+        /**
+         * Registers a slave host controller. A non-exclusive read lock is held for the duration of this registration
+         * process, blocking writes, but allowing the initiation of additional read locks.
```

Comment:
This should have a comment about how if the call succeeds, completeRegistration *must* be called.